crates/ty_python_semantic/resources/mdtest/snapshots/return_type.md_-_Function_return_type_-_Generator_functions.snap

```diff
@@ -0,0 +1,82 @@
+---
+source: crates/ty_test/src/lib.rs
+expression: snapshot
+---
+---
+mdtest name: return_type.md - Function return type - Generator functions
+mdtest path: crates/ty_python_semantic/resources/mdtest/function/return_type.md
+---
+
+# Python source files
+
+## mdtest_snippet.py
+
+```
+ 1 | import types
+ 2 | import typing
+ 3 | 
+ 4 | def f() -> types.GeneratorType:
+ 5 |     yield 42
+ 6 | 
+ 7 | def g() -> typing.Generator:
+ 8 |     yield 42
+ 9 | 
+10 | def h() -> typing.Iterator:
+11 |     yield 42
+12 | 
+13 | def i() -> typing.Iterable:
+14 |     yield 42
+15 | 
+16 | def j() -> str:  # error: [invalid-return-type]
+17 |     yield 42
+18 | import types
+19 | import typing
+20 | 
+21 | async def f() -> types.AsyncGeneratorType:
+22 |     yield 42
+23 | 
+24 | async def g() -> typing.AsyncGenerator:
+25 |     yield 42
+26 | 
+27 | async def h() -> typing.AsyncIterator:
+28 |     yield 42
+29 | 
+30 | async def i() -> typing.AsyncIterable:
+31 |     yield 42
+32 | 
+33 | async def j() -> str:  # error: [invalid-return-type]
+34 |     yield 42
+```
+
+# Diagnostics
+
+```
+error: lint:invalid-return-type: Return type does not match returned value
+  --> src/mdtest_snippet.py:16:12
+   |
+14 |     yield 42
+15 |
+16 | def j() -> str:  # error: [invalid-return-type]
+   |            ^^^ Expected `str`, found `types.GeneratorType`
+17 |     yield 42
+18 | import types
+   |
+info: Function is inferred as returning `types.GeneratorType` because it is a generator function
+info: See https://docs.python.org/3/glossary.html#term-generator for more details
+
+```
+
+```
+error: lint:invalid-return-type: Return type does not match returned value
+  --> src/mdtest_snippet.py:33:18
+   |
+31 |     yield 42
+32 |
+33 | async def j() -> str:  # error: [invalid-return-type]
+   |                  ^^^ Expected `str`, found `types.AsyncGeneratorType`
+34 |     yield 42
+   |
+info: Function is inferred as returning `types.AsyncGeneratorType` because it is a async generator function
```

Comment:
nit: should be "an async generator function" -- I know this is a bit of a pain to implement :/

Reply:
lol